### PR TITLE
CLN: Fully lint compat

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: statsmodels/compat/__init__.py [hunk 1]
```diff
@@ -1 +1,23 @@
-from .python import *
```
Comment (bashtage) [inline statsmodels/compat/__init__.py]: Can't you just leave this in and noqa it?  Or do you get other flake errors?

Comment (jbrockmendel) [reply statsmodels/compat/__init__.py]: That would be allowed.  In this case changing it surfaced the fact that `urlretrieve` was defined in the py3 section but not the py2 section.  Not that we actually need `urlretrieve`, but the explicitness can be useful.

**Comment by jbrockmendel** [comment]
@josef-pkt ping.  You can do this; I believe in you.

**Comment by jbrockmendel** [comment]
Looks like some unwanted stuff got in here in a rebase.  Since #5000 will make much of this unnecessary, closing.  Will re-do after #5000.

**Pull request closed without merging**